doc/userguide/upgrade.rst

```diff
@@ -82,6 +82,14 @@ Major changes
 - Unknown requirements in the ``requires`` keyword will now be treated
   as unmet requirements, causing the rule to not be loaded. See
   :ref:`keyword_requires`.
+- The configuration setting controlling stream checksum checks no longer affects
+  checksum keyword validation. Previously, when ``stream.checksum-validation``
+  was set to ``no``, the checksum keywords (e.g., ``ipv4-csum``, ``tcpv4-csum``, etc)
+  would either match or not match according to the value used with the checksum keyword.
+  Previous behavior would return a match when ``ipv4-csum: valid`` was specified and
+  not match if ``ipv4-csum: invalid`` was used. With 8.0, a match will occur based on the
+  computed checksum and the value (``valid`` or ``invalid``) agree.
```

Comment:
I'm not quite sure I understand this statement.

In 7, if I disable checksum validation, do the keywords always treat the checksum as OK? And now in 8 even if checksum validation is disabled, they will match on the actual checksum regardless if checksum-validation is disable or not at the stream level?

Reply:
In 7, if checksum validation is disabled, the csum rule keywords match based on the value used with the keyword. eg., `tcpv4-csum: invalid` wont't match; `tcpv4-csum: valid` matches

In 8, checksum-validation no longer affects the csum rule keywords. The csum rule keywords trigger if the checksum value (valid or invalid) match the value used with the rule keyword, e.g, `ipv4-csum: valid` (match if checksum is valid) or `ipv4-csum: invalid` match only when checksum is invalid)

Reply:
I find this bit confusing:

> In 7, if checksum validation is disabled, the csum rule keywords match based on the value used with the keyword.

Maybe something like?

> In 7, if checksum validation is disabled, the rule keywords checking for checksum will always consider it valid. Meaning `tcpv3-csum: invalid` will never match.

> In 8, checksum-validation no longer affects the csum rule keywords. `ipv4-csum: valid` only match if the checksum is valid, even if engine checksum validations have been disabled.

Or something like that..  In particular, this is the bit I find confusing:

> The csum rule keywords trigger if the checksum value (valid or invalid) match the value used with the rule keyword